test/integrations/destinations/facebook_pixel/processor/ecommTestData.ts

```diff
@@ -0,0 +1,705 @@
+  products: [
+    {
+      product_id: '017c6f5d5cf86a4b22432066',
+      sku: '8732-98',
+      name: 'Just Another Game',
+      price: 22,
+      position: 2,
+      category: 'Games and Entertainment',
+      url: 'https://www.myecommercewebsite.com/product',
+      image_url: 'https://www.myecommercewebsite.com/product/path.jpg',
+    },
+    {
+      product_id: '89ac6f5d5cf86a4b64eac145',
+      sku: '1267-01',
+      name: 'Wrestling Trump Cards',
+      price: 4,
+      position: 21,
+      category: 'Card Games',
+    },
+  ],
+  category: 'dummy',
+  quantity: 10,
+  revenue: 100,
+  price: 50,
+  product_id: '12345',
+  order_id: '23456',
+};
+const commonTimestamp = new Date('2023-10-14');
+const commonStatTags = {
+  errorCategory: 'dataValidation',
+  errorType: 'instrumentation',
+  destType: 'FACEBOOK_PIXEL',
+  module: 'destination',
+  implementation: 'native',
+  feature: 'processor',
+};
+
+export const ecommTestData: ProcessorTestData[] = [
+  {
+    id: 'facebook_pixel-test-1',
+    name: 'facebook_pixel',
+    description:
+      'Track call : product list viewed event call with properties without product array',
+    scenario: 'ecommerce',
+    successCriteria:
+      'It should be internally mapped to ViewContent, with necessary mapping from message.properties and should be sent to the destination',
+    feature: 'processor',
+    module: 'destination',
+    version: 'v0',
+    input: {
+      request: {
+        body: [
+          {
+            message: generateTrackPayload({
+              event: 'product list viewed',
+              properties: commonPropertiesWithoutProductArray,
+              context: {
+                traits: commonUserTraits,
+              },
+              timestamp: commonTimestamp,
+            }),
+            metadata: generateMetadata(1),
+            destination: commonDestination,
+          },
+        ],
+      },
+    },
+    output: {
+      response: {
+        status: 200,
+        body: [
+          {
+            output: transformResultBuilder({
+              version: '1',
+              type: 'REST',
+              method: 'POST',
+              endpoint: `https://graph.facebook.com/${VERSION}/dummyPixelId/events?access_token=09876`,
+              headers: {},
+              params: {},
+              FORM: {
+                data: [
+                  '{"user_data":{"external_id":"3ffc8a075f330402d82aa0a86c596b0d2fe70df38b22c5be579f86a18e4aca47","em":"48ddb93f0b30c475423fe177832912c5bcdce3cc72872f8051627967ef278e08","client_user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:84.0) Gecko/20100101 Firefox/84.0"},"event_name":"ViewContent","event_time":1697241600,"event_id":"12345","action_source":"website","custom_data":{"category":"dummy","quantity":10,"value":100,"product_id":"12345","content_ids":["dummy"],"content_type":"product_group","contents":[{"id":"dummy","quantity":1}],"content_category":"dummy","currency":"USD"}}',
```

Comment:
Difficult to read this as string like I mentioned if we can use JSON.stringify

Reply:
On top of a json object

Reply:
done